test/integration/etl_test.py

```diff
@@ -17,13 +17,21 @@
 logger = logging.getLogger(__name__)
 
 
-def test_pudl_engine(pudl_engine):
-    """Try creating a pudl_engine...."""
+def test_pudl_engine(pudl_engine, pudl_sql_io_manager, check_foreign_keys):
+    """Get pudl_engine and do basic inspection.
+
+    By default the foreign key checks are not enabled in pudl.sqlite. This test will
+    check if there are any foregin key errors if check_foreign_keys is True.
+    """
     assert isinstance(pudl_engine, sa.engine.Engine)  # nosec: B101
     insp = sa.inspect(pudl_engine)
     assert "plants_pudl" in insp.get_table_names()  # nosec: B101
     assert "utilities_pudl" in insp.get_table_names()  # nosec: B101
 
+    if check_foreign_keys:
+        # Raises ForeignKeyErrors if there are any
+        pudl_sql_io_manager.check_foreign_keys()
+
```

Comment:
Now that we have this check we can delete `test/validate/database_test.py`. We could also do it in another PR so we don't have to rerun the CI for this PR.